### ISSUE TITLE
Kernel panic booting RX 5700 XT

### ISSUE THREAD

**Issue by pieter91** [iGPU/AMD_OC_Ryzen#6]
I understand from reading the thread on amd-osx.com that you won't be working on this repo much anymore. I am however having some trouble with your latest commits and I'd be grateful for some help. 

I get either an mp_rv_action kernel panic, or a general protection panic with X6000 kext in backtrace. I've already tried disabling WhateverGreen.kext, SSDT-X570-RX5700XT-OptPowerTable-slot-1.aml, AGPMInjector-MacPro7,1-RX5700XT.kext, and using the non-OptPowerTable SSDT. 

Any ideas on what else I could try?

**Comment by pieter91** [comment]
By the way, switching back to OpenCore v0.5.7 and disabling WhateverGreen does not seem to fix the cursor glitches for me.

**Comment by iGPU** [comment]
I've not seen any "mp_rv_action kernel panic" errors, and I'm using same v058 on X299, Z390 and X570 (GB Aorus Master) mobo.

It sounds like possibly a problem with Lilu. I'd updated it to the latest commit, but maybe restore older version to see if that helps. For testing, I'd not recommend using any PowerTable injection until everything else is stable. The AGPM file should be fine.

If v057 and no WEG didn't help your glitches, then you need to look at your BT/WiFi setup. Did you swap card or using PCIe (and which one)? Are you using the supplied kext files or trying to go 'native' and use none? If the latter, try enabling all of the supplied BT/WiFi kext files.

Are you using Mojave or Catalina, and which version?

**Comment by pieter91** [comment]
I'm on Catalina 10.15.4 (supplemental update). I've got an Apple BCM94360CD in a PCIe adapter. It never needed anything to make it work in macOS on my old Haswell system. I also noticed the MacProMemoryNotificationDisabler kext causing kernel panics without WhateverGreen, so that might also be a factor. 

I'll check on Lilu then, and try using your kexts for wireless. I'll report back here tomorrow. 

Could physically removing the Intel M.2 wireless card make a difference?

Thanks for your help!

**Comment by Logicwax** [comment]
I'm getting the same thing.   why won't this repo continue being maintained?

**Comment by iGPU** [comment]
pieter91,

Please read the GitHub notes carefully. The 2 large PCIe slots farthest from CPU share lanes with the M.2 slots and affect where the M.2 slots appear. I found that BT/WiFi was not consistent without the kext files (despite everyone being so consumed by 'native' behavior'; there's nothing native about any Hackintosh). The most ideal I found was swapping out the builtin BT/Wifi card and avoiding another PCIe interaction.

Personally, I've grown to dislike the X570 chip and is annoying lane sharing. If you activate TB (again, read the notes), the lane sharing is further disturbed. On the Intel side, SATA slots are shared with M.2 slots, not PCIe slots. This is easier to deal with as both are SSD storage.

If you're using MacProMemoryNotificationDisabler then you're using MacPro7,1. I think this is a mistake and only iMacPro1,1 should be used. You're creating more problems with MacPro7,1. i have no idea why everyone is fixated on this SMBIOS. None of our builds are remotely similar.

**Comment by iGPU** [comment]
Logicwax,

If you read the original thread that led to this repository, you'll see that I'm no longer using the ASRock Creator. I now have, for a short time, a GB X570 Aorus Master. Both are very similar.

Tonight, I uploaded major changes to the repository, including re-writing all dGPU SSDT files, adding RadeonBooster kext (which itself automatically injects the correct AGPM). This new kext not only improves dGPU functioning, but also allowed me to remove the discrete and confusing AGPMInjector kexts, greatly simplifying this section of the EFI.

Also, OC had a new incremental v058 update that provides an Auxiliary menu item (press the space bar when the menu items appear): Shutdown.

**Comment by pieter91** [comment]
Just a quick update. I managed to get my system booting macOS with OpenCore 0.5.8 by disabling WhateverGreen and enabling RadeonBoost. For my silly MacPro7,1 SMBIOS I've had to disable MacProMemoryNotificationDisabler.kext, but it's a minor annoyance, so I can live without. Beats having to log out and log in to iCloud all over again to switch SMBIOS. I'll leave that for another day. 
I've noticed not every boot is successful. I've had the occasional kernel panic (as before) at the end of booting, or right afterwards. Once it's running, most things are back to how they were before. But now without the cursor stuttering. 
I checked hardware h.264 and h.265 encoding and decoding, but only h.264 is enabled. That's the downside of missing WhateverGreen I suppose.
